File: q2264/q2264.java
class Solution {
    public String largestGoodInteger(String num) {
        StringBuilder result = new StringBuilder();
        for (int i = 2; i < num.length(); i++) {
            if (num.charAt(i - 2) == num.charAt(i) && num.charAt(i - 1) == num.charAt(i)) {
                if (result.length() == 0 || num.charAt(i) > result.charAt(0)) {
                    result = num.substring(i - 2, i + 1);
                }
            }
        }
        return result;
    }
}

File: q2264/q2264.cpp
class Solution
{
public:
    string largestGoodInteger(string num)
    {
        string result = "";
        for (int i = 2; i < num.size(); i++)
        {
            if (num[i - 2] == num[i] && num[i - 1] == num[i])
            {
                if (result.size() == 0 || result[0] < num[i])
                {
                    result = string(3, num[i]);
                }
            }
        }
        return result;
    }
};